sqlglot/dialects/duckdb.py

```diff
@@ -738,3 +738,31 @@ def withingroup_sql(self, expression: exp.WithinGroup) -> str:
             this = self.sql(expression, "this").rstrip(")")
 
             return f"{this}{expression_sql})"
+
+        def length_sql(self, expression: exp.Length) -> str:
+            arg = expression.this
+
+            if not arg.type:
+                from sqlglot.optimizer.annotate_types import annotate_types
+
+                arg = annotate_types(arg)
+
+            if arg.is_type(exp.DataType.Type.VARCHAR) or arg.is_string:
+                return self.func("LENGTH", arg)
```

Comment:
Let's flip order of conditions and do `is_type(*exp.DataType.TEXT_TYPES)` instead